test/parallel/market.fee.scenarios.test.ts

```diff
@@ -0,0 +1,973 @@
+    const poolEvent = await Sudo.batchAsSudoFinalized(
+      ...(await addTestExtrinsic(
+        [firstCurrency, secondCurrency],
+        [true, "StableSwap"],
+        [true, "Xyk"],
+        [true, "Xyk"],
+      )),
+      Assets.mintToken(firstCurrency, testUser, threshold.muln(2)),
+    );
+    await updateFeeLockMetadata(sudo, null, null, null, [
+      [firstCurrency, true],
+    ]);
+
+    const liqId = await getPoolIdFromEvent(poolEvent);
+
+    const tokenValueBefore = await getTokensAccountInfo(
+      testUser.keyRingPair.address,
+      firstCurrency,
+    );
+
+    const buyPrice = stringToBN(
+      JSON.parse(
+        JSON.stringify(
+          await api.rpc.market.calculate_buy_price(
+            liqId,
+            secondCurrency,
+            threshold.muln(3).divn(2),
+          ),
+        ),
+      ),
+    );
+
+    await swapTokenAndReceiveSlippageError(
+      testUser,
+      [firstCurrency, secondCurrency],
+      liqId,
+      swapAmount,
+      xykErrors.ExcessiveInputAmount,
+      "Buy",
+    );
+
+    const tokenValueAfter = await getTokensAccountInfo(
+      testUser.keyRingPair.address,
+      firstCurrency,
+    );
+    const tokenDiff = stringToBN(tokenValueBefore.free).sub(
+      stringToBN(tokenValueAfter.free),
+    );
+
+    expect(tokenDiff).bnEqual(buyPrice.divn(1000).muln(3).addn(2));
+  });
+
+  test("GIVEN buyAsset AND amount > threshold AND fail on slippage THEN operation fails and we take fee", async () => {
+    await Sudo.batchAsSudoFinalized(
+      ...(await addTestExtrinsic(
+        [firstCurrency, secondCurrency],
+        [true, "Xyk"],
+        [true, "Xyk"],
+        [true, "Xyk"],
+      )),
+      Assets.mintToken(firstCurrency, testUser, threshold.muln(2)),
+    );
+    await updateFeeLockMetadata(sudo, null, null, null, [
+      [firstCurrency, true],
+    ]);
+
+    const liqId = await getLiquidityAssetId(firstCurrency, secondCurrency);
+    const poolBalance = await getBalanceOfPool(firstCurrency, secondCurrency);
+    const buyPrice = calculate_buy_price_local(
+      poolBalance[0],
+      poolBalance[1],
+      swapAmount,
+    );
+
+    const tokenValueBefore = await getTokensAccountInfo(
+      testUser.keyRingPair.address,
+      firstCurrency,
+    );
+
+    await swapTokenAndReceiveSlippageError(
+      testUser,
+      [firstCurrency, secondCurrency],
+      liqId,
+      swapAmount,
+      xykErrors.ExcessiveInputAmount,
+      "Buy",
+    );
+
+    const tokenValueAfter = await getTokensAccountInfo(
+      testUser.keyRingPair.address,
+      firstCurrency,
+    );
+    const tokenDiff = stringToBN(tokenValueBefore.free).sub(
+      stringToBN(tokenValueAfter.free),
+    );
+    expect(tokenDiff.divn(10e6)).bnEqual(
+      buyPrice.divn(1000).muln(3).divn(10e6),
+    );
+  });
+});
+
+describe("MultiSwap scenarios with slippage error, user has only sold asset", () => {
```

Comment:
we are not reproducing the exact scenario, plz whitelist and do multiswaps, not singleswaps

Reply:
Done